src/Streamly/Internal/Data/Stream/StreamK.hs

```diff
@@ -751,19 +751,8 @@ mapMaybe f = go
 --
 -- @since 0.1.0
 {-# INLINE zipWith #-}
-zipWith :: (a -> b -> c) -> Stream m a -> Stream m b -> Stream m c
-zipWith f = go
-
-    where
-
-    go mx my = mkStream $ \st yld sng stp -> do
-        let merge a ra =
-                let single2 b = sng (f a b)
-                    yield2 b rb = yld (f a b) (go ra rb)
-                 in foldStream (adaptState st) yield2 single2 stp my
-        let single1 a = merge a nil
-            yield1 = merge
-        foldStream (adaptState st) yield1 single1 stp mx
+zipWith :: Monad m => (a -> b -> c) -> Stream m a -> Stream m b -> Stream m c
```

Comment:
This required adding a `Monad` constraint here.

Reply:
This is an internal change only, so should be fine in minor release.